tower/src/builder/mod.rs

```diff
@@ -688,6 +688,121 @@ impl<L> ServiceBuilder<L> {
     {
         self
     }
+
+    /// Convert the resulting service into a [`Service`] + [`Send`] trait object.
+    ///
+    /// This applies the [`BoxService:layer()`] layer. See that method for more details.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use tower::{Service, ServiceBuilder, BoxError, util::BoxService};
+    /// use std::time::Duration;
+    /// #
+    /// # struct Request;
+    /// # struct Response;
+    /// # impl Response {
+    /// #     fn new() -> Self { Self }
+    /// # }
+    ///
+    /// let service: BoxService<Request, Response, BoxError> = ServiceBuilder::new()
+    ///     .boxed()
+    ///     .load_shed()
+    ///     .concurrency_limit(64)
+    ///     .timeout(Duration::from_secs(10))
+    ///     .service_fn(|req: Request| async {
+    ///         Ok::<_, BoxError>(Response::new())
+    ///     });
+    /// # let service = assert_service(service);
+    /// # fn assert_service<S, R>(svc: S) -> S
+    /// # where S: Service<R> { svc }
+    /// ```
+    ///
+    /// [`BoxService:layer()`]: crate::util::BoxService::layer()
+    #[cfg(feature = "util")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "util")))]
+    pub fn boxed<S, R>(
+        self,
+    ) -> ServiceBuilder<
+        Stack<
+            tower_layer::LayerFn<
+                fn(
+                    L::Service,
+                ) -> crate::util::BoxService<
+                    R,
+                    <L::Service as Service<R>>::Response,
+                    <L::Service as Service<R>>::Error,
+                >,
+            >,
```

Comment:
I went down that route initially actually but you end up having to have the request type as a type param on the layer:

```rust
// have to have `T` here otherwise its unconstrained in the layer impl
pub struct CloneBoxServiceLayer<T> {
    _request: PhantomData<T>,
}

impl<S, T> Layer<S> for CloneBoxServiceLayer<T>
where
    S: Service<T> + Clone + Send + 'static,
    S::Future: Send + 'static,
{
    type Service = CloneBoxService<T, S::Response, S::Error>;

    fn layer(&self, inner: S) -> Self::Service {
        CloneBoxService::new(inner)
    }
}
```

That could be workable to leads to inference issues when actually using the builder method:

```
---- src/builder/mod.rs - builder::ServiceBuilder<L>::clone_boxed (line 754) stdout ----
error[E0284]: type annotations needed
  --> src/builder/mod.rs:765:6
   |
14 |     .clone_boxed()
   |      ^^^^^^^^^^^ cannot infer type for type parameter `S`
   |
   = note: cannot satisfy `<_ as Service<Request>>::Future == _`
```

Reply:
ugh, never mind. it's probably better to avoid introducing phantomdata anyway, as they will result in rustc having to typecheck a larger type, potentially impacting build time... this is fine, then.